src/session/session_compact.c

```diff
@@ -260,17 +262,13 @@ __wt_session_compact_check_interrupted(WT_SESSION_IMPL *session)
 
 /*
  * __compact_checkpoint --
- *     This function does wait and force checkpoint.
+ *     This function waits and triggers a checkpoint that can be forced.
  */
 static int
-__compact_checkpoint(WT_SESSION_IMPL *session)
+__compact_checkpoint(WT_SESSION_IMPL *session, bool force)
 {
-    /*
-     * Force compaction checkpoints: we don't want to skip it because the work we need to have done
-     * is done in the underlying block manager.
-     */
     const char *checkpoint_cfg[] = {
-      WT_CONFIG_BASE(session, WT_SESSION_checkpoint), "force=1", NULL};
+      WT_CONFIG_BASE(session, WT_SESSION_checkpoint), force ? "force=1" : NULL, NULL};
```

Comment:
This isn't quite what I was imagining, I was hoping there was a way we could make the checkpoint on the particular object being compacted have `force=1`, rather than leave `force=1` enabled for all tables on the final checkpoint that is part of compact.

In thinking about this, I looked at the history of the code and came across an interesting conversation here:
https://jira.mongodb.org/browse/WT-2969?focusedCommentId=1410192&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-1410192
I wonder if that's relevant to background compaction? I also wonder if it is a pointer to a broader issue - should checkpoint in service of compact give up if it's blocking other things for too long?

Reply:
@agorrod, could we have a `force` boolean at the btree level that would be set by compaction?

> should checkpoint in service of compact give up if it's blocking other things for too long?

This is definitely something we should look into. There are multiple avenues to explore:

- A checkpoint issued by compaction should only checkpoint what compaction modified (this ticket)
- It would be great to give compaction "a goal" instead of trying to keep compacting as much as possible as it can generate too much IO and it will give more work to the checkpoint. In theory, large tables could be easily affected.
- Finally, timewise, there is a `timeout` field for compaction itself and for checkpoints it is a bit tricky, we cannot give up. If a checkpoint is slow (whether it is issued from compaction or not), I suspect this is either due to lots of things to be checkpointed or something else slowing it down.

Regarding the JIRA comment, I hope I understand it right. This could be super interesting if background compaction does not trigger the checkpoints and we let the application do it. cc: @Sean04

Reply:
I will continue the conversation on the JIRA ticket.